src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/SpecialParametersBindingMetadataProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Metadata
+{
+    public class SpecialParametersBindingMetadataProvider : IBindingMetadataProvider
```

Comment:
`SpecialBindingSourceMetadataProvider`

In general **always** avoid plurals like `SpecialParameters` in names.